.circleci/config.yml

```diff
@@ -94,6 +94,11 @@ jobs:
           working_directory: ~/build_dir
           command: |
              tar Jxf ~/sdk/$SDK_FILE --strip=1
+             touch .config
+             make prepare-tmpinfo scripts/config/conf
+             ./scripts/config/conf --defconfig=.config Config.in
+             make prereq
+             rm .config
```

Comment:
(Apologies for the silly questions.) Just want to clarify: the current CI build errors are because python3 isn't part of the Docker image, and this would be fixed by using v1.0.4 of the Docker image, yes?

The purpose of this code is to run `make prereq` separately (and before it is normally called), so that any errors that may occur (in the future) will be viewable in the log?

Reply:
Yes, exactly

Reply:
Current [CI build issue](https://circleci.com/gh/openwrt/packages/4636) visible with `make prereq`:

```
Checking 'file'... ok.
Checking 'ldconfig-stub'... ok.

Build dependency: Please install Python >= 3.5
Build dependency: Please install Python >= 3.5

/home/build/build_dir/include/prereq.mk:12: recipe for target 'prereq' failed
Prerequisite check failed. Use FORCE=1 to override.
/home/build/build_dir/include/toplevel.mk:167: recipe for target 'staging_dir/host/.prereq-build' failed
make[1]: *** [staging_dir/host/.prereq-build] Error 1
/home/build/build_dir/include/toplevel.mk:83: recipe for target 'prepare-tmpinfo' failed
```

Reply:
I see... Actually I don't think we should do this here. This only tests one part of the SDK build process, inside of CI for packages.

I'm in favour of CI for the SDK, but it should be a separate, dedicated, complete build test for the SDK only. (Building a known-good package, for example.) I assume the ongoing Docker/CI work will cover this.

I'm ok with leaving this in for now (a code comment about it's purpose would be nice), but this isn't the best solution for diagnosing/solving SDK problems.

Reply:
> I'm in favour of CI for the SDK, but it should be a separate, dedicated, complete build test for the SDK only. (Building a known-good package, for example.) I assume the ongoing Docker/CI work will cover this.

I would say, that it's already very well covered, all the packages are build using the SDK already.

> (a code comment about it's purpose would be nice)

git log -L97,101:.circleci/config.yml

> but this isn't the best solution for diagnosing/solving SDK problems

It's a problem of Docker image and the script used in CI, it's not the problem of the SDK.

Reply:
CI broke because of openwrt/openwrt@200f2666fb1c8d9d128824dc5586e0e66386971f, yes?

This code isn't for testing packages; it's for testing if the Docker image / SDK is working correctly, before testing packages. I hope this difference is observable.

If we want to test if the Docker image / SDK is working correctly, before testing packages, then I am suggesting we do an additional, more comprehensive test.

Reply:
> CI broke because of openwrt/openwrt@200f266, yes?

No, CI was in broken state right after the switch to python3, openwrt/openwrt@200f266 made it just visible.

Reply:
Ok but do you understand my point?